pkg/api/customization/clustertemplate/cluster_template.go

```diff
@@ -0,0 +1,124 @@
+	resource.AddAction(apiContext, "disable")
+	resource.Links["revisions"] = apiContext.URLBuilder.Link("revisions", resource)
+}
+
+func (w Wrapper) ActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	var template client.ClusterTemplate
+	if err := access.ByID(apiContext, &managementschema.Version, client.ClusterTemplateType, apiContext.ID, &template); err != nil {
+		return err
+	}
+	switch actionName {
+	case "disable":
+		obj, err := w.ClusterTemplates.GetNamespaced(namespace.GlobalNamespace, template.Name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if obj.Spec.Disabled {
+			return nil
+		}
+		toUpdate := obj.DeepCopy()
+		toUpdate.Spec.Disabled = true
+
+		_, err = w.ClusterTemplates.Update(toUpdate)
+		if err != nil {
+			return err
+		}
+		apiContext.WriteResponse(http.StatusNoContent, map[string]interface{}{})
+		return nil
+	default:
+		return fmt.Errorf("bad action for clusterTemplate %v", actionName)
+	}
+}
+func (w Wrapper) LinkHandler(apiContext *types.APIContext, next types.RequestHandler) error {
+	switch apiContext.Link {
+	case "revisions":
+		logrus.Infof("Revision link invoked")
+		var template client.ClusterTemplate
+		if err := access.ByID(apiContext, &managementschema.Version, client.ClusterTemplateType, apiContext.ID, &template); err != nil {
+			return err
+		}
+		logrus.Infof("template %v", template)
+		conditions := []*types.QueryCondition{
+			types.NewConditionFromString(client.ClusterTemplateRevisionFieldClusterTemplateID, types.ModifierEQ, []string{template.ID}...),
+		}
+		var templateVersions []map[string]interface{}
+		if err := access.List(apiContext, &managementschema.Version, client.ClusterTemplateRevisionType, &types.QueryOptions{Conditions: conditions}, &templateVersions); err != nil {
+			return err
+		}
+		sort.SliceStable(templateVersions, func(i, j int) bool {
+			val1, err := time.Parse(time.RFC3339, convert.ToString(values.GetValueN(templateVersions[i], "created")))
+			if err != nil {
+				logrus.Infof("error parsing time %v", err)
+			}
+			val2, err := time.Parse(time.RFC3339, convert.ToString(values.GetValueN(templateVersions[j], "created")))
+			if err != nil {
+				logrus.Infof("error parsing time %v", err)
+			}
+			return val1.After(val2)
+		})
+		apiContext.Type = client.ClusterTemplateRevisionType
+		apiContext.WriteResponse(http.StatusOK, templateVersions)
+		return nil
+	}
+	return nil
+}
+
+func (w Wrapper) RevisionActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	var templateRevision client.ClusterTemplateRevision
+	if err := access.ByID(apiContext, &managementschema.Version, client.ClusterTemplateType, apiContext.ID, &templateRevision); err != nil {
+		return err
+	}
+	switch actionName {
+	case "disable":
+		obj, err := w.ClusterTemplateRevisions.GetNamespaced(namespace.GlobalNamespace, templateRevision.Name, v1.GetOptions{})
```

Comment:
After talking to @prachidamle offline, found out that the intent of the action to turn on the flag was to prevent updates on all other fields. We can achieve similar functionality by setting noUpdate norman flag in rancher/types.

Reply:
@alena1108 on second thoughts, maybe we should allow updates on ClusterTemplateRevision fields, unless it is being used by some clusters?